chapter_11_cqrs.asciidoc

```diff
@@ -139,7 +139,7 @@ obj-...
 
 
 [[views_dot_py]]
-.Views do... raw sql??? (src/allocation/views.py)
+.Views do... raw SQL??? (src/allocation/views.py)
```

Comment:
I know you're attached to this bait-and-switch, but I honestly don't think it works very well for explaining CQRS. It's not just the exuberant language, I think that structurally it's unhelpful. I think it would be better to leave the raw SQL until after the SQLAlchemy query example.

More fundamentally, though, there's a definite sense in the prose that we, the authors, think this is a stupid thing that nobody should do, but I don't think that. I think that if you don't get performance issues from using your domain model to service reads, then fine but if you're struggling to build complex reports, or if you're doing high-volume reads on a system with a complex domain model, then you should think again.

Maybe we should explain why splitting into a new datastore is helpful? The hardest part of a system to scale is the data store because you can't (usually) scale it horizontally. CQRS solves that problem at a stroke by giving us as many read-replicas as we like. Since reads usually dominate DB traffic by an order of magnitude, that makes it much easier for us to scale.

Maybe part of the problem here is that we don't have a good motivating problem, so we're trying to shoe-horn this simple requirement into a solution that's overpowered.

Reply:
hmmm. i am finding it hard to let go of that bait-and-switch, you're right.  i'm hoping it's not quite as fatal a flaw as you think?  (i mean, i kinda stole it from you and your cqrs blog post lol)

re: the sense that we think this is a stupid thing that nobody should do, let's definitely try and get rid of that.  i think actually the example in the chapter illustrate the fact that "just' using the domain model to service read can be suboptimal, that shows up quite nicely as soon as we show the `uow.products.get` example with its three layers of python looping.

very much like the bumf on scaling.  let's add some of that.

shall i try and rewrite the intro up to the bait-and-switch and see if it gets sane enough? if it doesn't work we'll rethink the structure, and i should probably let go of the chapter for a bit.